cdap-api/src/main/java/co/cask/cdap/internal/dataset/DatasetRuntimeContext.java

```diff
@@ -24,13 +24,15 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.Closeable;
+import java.io.IOException;
 import java.lang.annotation.Annotation;
 import javax.annotation.Nullable;
 
 /**
  * The internal context object used by the Dataset framework to track Dataset methods.
  */
-public abstract class DatasetRuntimeContext {
+public abstract class DatasetRuntimeContext implements Closeable {
```

Comment:
Suggest to extends from `AutoCloseable` and override the `close` method to not throw any exception.